source/fundamentals/connections/connection-options.txt

```diff
@@ -0,0 +1,197 @@
+
+   * - **maxPoolSize**
+     - integer
+     - ``100``
+     - Specifies the maximum number of connections that a connection pool may
+       have at a given time.
+
+   * - **replicaSet**
+     - string
+     - ``null``
+     - Specifies the replica set name for the cluster. All nodes in the
+       replica set must have the same replica set name, or the Client will not
+       consider them as part of the set.
+
+   * - **maxIdleTimeMS**
+     - integer
+     - ``0``
+     - Specifies the maximum amount of time a connection can remain idle
+       in the connection pool before being removed and closed. The
+       default is ``0``, meaning a connection can remain unused
+       indefinitely.
+
+   * - **minPoolSize**
+     - integer
+     - ``0``
+     - Specifies the minimum number of connections that the driver maintains
+       in a single connection pool.
+
+   * - **serverSelectionTimeoutMS**
+     - integer
+     - ``30000``
+     - Specifies the number of milliseconds to wait to find an available,
+       suitable server to execute an operation.
+
+   * - **heartbeatFrequencyMS**
+     - integer
+     - ``10000``
+     - Specifies the number of milliseconds to wait between periodic
+       background server checks.
+
+   * - **tls**
+     - boolean
+     - ``false``
+     - Specifies whether to establish a Transport Layer Security (TLS)
+       connection with the instance. This is automatically set to ``true``
+       when using a DNS seedlist (SRV) in the connection string. You can
+       override this behavior by setting the value to ``false``.
+
+   * - **w**
+     - string or integer
+     - ``null``
+     - Specifies the write concern. To learn more about values, see the
+       server documentation on
+       :manual:`Write Concern options </reference/write-concern>`.
+
+   * - **directConnection**
+     - boolean
+     - ``false``
+     - Specifies whether to force dispatch **all** operations to the host
+       specified in the connection URI.
+
+For a full list of connection options, see the `ClientOptions API
+documentation
+<{+api+}/mongo/options#ClientOptions>`__.
+
+.. _golang-timeout-setting:
+
+Single Timeout Setting
+----------------------
+
+You can set a single ``Timeout`` option on your ``Client`` instance to
+specify the amount of time that a single operation can take to execute.
+Set a timeout by using the ``SetTimeout()`` method or specifying the
+``timeoutMS`` option in your connection URI string. ``Database``, ``Collection``,
+``Session``, ``ChangeStream``, and ``Bucket`` instances elsewhere in
+your code inherit the ``Timeout`` option from ``Client`` if you do not
+set a different timeout in a Context for specific operations.
+
+If you pass a Context into an operation with a deadline, the driver uses
+that Context deadline for the operation. If the context does not have a
+deadline, the driver derives a new Context from the given Context by using
+the ``Timeout`` option set on the ``Client``.
```

Comment:
This might be due to me not being very familiar with Go, but this paragraph is confusing to me. It seems a bit jarring to be talking about setting timeouts, but then the next paragraph is talking about using Context deadlines. Is a "deadline" the same thing as a timeout? If so I think we should stick with that wording. Happy to workshop some ways to rephrase this once I have a better understanding of what it's trying to say.

Reply:
Will clean this up (old text)